### PR TITLE
Handle pypi data fetch

### PR DESCRIPTION
**Pull request by manasaV3**
## Description
Relates to https://github.com/chanzuckerberg/napari-hub/issues/861
Depends on https://github.com/chanzuckerberg/napari-hub/pull/1061

Changes introduced:
- Updates the plugin update process
   - Fetches all the plugins that are currently marked latest from the plugin dynamo table.
   - Gets all the latests versions of plugins from pypi
   - For all plugins and version in pypi but not in the data fetched from dynamo, we fetch the new data
   - For plugins fetched from dynamo but not in data fetched from pypi, we mark them to no longer be the latest.

### CODE REVIEW

**Review by aganders3 — CHANGES_REQUESTED** [review]
Sorry for the number of comments - as mentioned many of these things are nitpicks. Overall this code is very readable and makes sense to me! There are a few things I would like to see changed as I think they could hide errors down the line in both testing and usage (for example the mutable fixtures, `MockResponse`, and `get_attribute`).

File: napari-hub-commons/src/nhcommons/models/plugin_metadata_type.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
     def __new__(cls):
         return object.__new__(cls)
 
     PYPI = ()
     SOURCE_CONTROL = ()
     DISTRIBUTION = ()
+
+    def to_version_type(self, version):
+        return f"{version}:{self.name}"
```
Comment (aganders3) [inline napari-hub-commons/src/nhcommons/models/plugin_metadata_type.py:14]: I don't think this implementation is correct - the resulting enum only has the `PYPI` member and the others are kind of like aliases for it.

```python
>>> PluginType.PYPI.name
'PYPI'
>>> PluginType.SOURCE_CONTROL.name
'PYPI'
```

Maybe you want to use `auto` to set the member values:
```
from enum import auto
class PluginType(Enum):
     PYPI = auto()
     SOURCE_CONTROL = auto()
     DISTRIBUTION = auto()
```

File: data-workflows/plugin/processor.py [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+    # update for removed plugins and existing older version of plugins
+    for name, plugin in dynamo_latest_plugins.items():
+        if pypi_latest_plugins.get(name) != plugin.version:
+            put_pypi_record(
+                plugin=name, version=plugin.version, is_latest=False
+            )
+
+
+def _update_for_new_plugin(plugin_name: str, version: str):
+    put_pypi_record(plugin=plugin_name, version=version, is_latest=True)
```
[the remark this answers is not in the document]
Comment (aganders3) [reply data-workflows/plugin/processor.py]: Sorry I think I'm just missing some of the context here. My initial comment was about this being is a 1-line function, and the inner function itself (`put_pypi_record`) is used directly in the code right below where this gets called. If the function will grow I think it makes sense, but then to be consistent I might suggest abstracting both calls in the code above.

Re: `pass` I would reserve it for where necessary and instead put a `# TODO` or something here.

File: napari-hub-commons/src/nhcommons/_tests/models/test_plugin.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class TestPlugin:
+
+    @pytest.fixture()
+    def plugin_table(self, aws_credentials):
+        with mock_dynamodb():
+            yield create_dynamo_table(plugin._Plugin, 'plugin')
+
+    @classmethod
+    def _put_item(cls, table, name, version, is_latest):
```
[the remark this answers is not in the document]
Comment (aganders3) [reply napari-hub-commons/src/nhcommons/_tests/models/test_plugin.py:15]: My thoughts on this are just that I prefer not to take the `cls` parameter if it is not used. I think it makes it more clear by reading the signature what you can expect in the method.

**Comment by manasaV3** [comment]
Changes from this PR have been merged into the main as a part of https://github.com/chanzuckerberg/napari-hub/pull/1108

**Pull request closed without merging**